model_templates/inference/python3_pytorch_multiclass/custom.py

```diff
@@ -0,0 +1,115 @@
+import os
+import pickle
+from typing import List, Optional, Any
+
+import numpy as np
+import pandas as pd
+from sklearn.preprocessing import LabelEncoder
+import torch
+
+from preprocessing import dense_preprocessing_pipeline
+from model_utils import build_classifier, train_classifier, save_torch_model
+
+preprocessor = None
+
+
+def load_model(code_dir: str) -> Any:
+    """
+    Can be used to load supported models if your model has multiple artifacts, or for loading
+    models that **drum** does not natively support
+
+    Parameters
+    ----------
+    code_dir : is the directory where model artifact and additional code are provided, passed in
+
+    Returns
+    -------
+    If used, this hook must return a non-None value
+    """
+    global preprocessor
+    with open(os.path.join(code_dir, "preprocessor.pkl"), mode="rb") as f:
+        preprocessor = pickle.load(f)
+
+    model = torch.load(os.path.join(code_dir, "artifact.pth"))
+    model.eval()
+    return model
+
+
+def transform(data: pd.DataFrame, model: Any) -> pd.DataFrame:
+    """
+    Intended to apply transformations to the prediction data before making predictions. This is
+    most useful if **drum** supports the model's library, but your model requires additional data
+    processing before it can make predictions
+
+    Parameters
+    ----------
+    data : is the dataframe given to **drum** to make predictions on
+    model : is the deserialized model loaded by **drum** or by `load_model`, if supplied
+
+    Returns
+    -------
+    Transformed data
+    """
+    if preprocessor is None:
+        raise ValueError("Preprocessor not loaded")
+
+    return preprocessor.transform(data)
+
+
+def fit(
```

Comment:
why is there a fit method in the inference template though.

Reply:
the model was fit with drum. figured it made sense to keep

Reply:
But what about all the other inference templates

Reply:
none of them were fit with drum. they predate drum even being able to fit. I think most of them straight up predate drum

Reply:
So why do we have inference and training models templates then?

Reply:
the other inference templates were not made with drum, but work with drum for scoring.